hibernate-core/src/main/java/org/hibernate/metamodel/model/domain/internal/DomainModelHelper.java

```diff
@@ -29,7 +29,7 @@
 		}
 
 		// first, try to find it by name directly..
-		ManagedDomainType<S> subManagedType = jpaMetamodel.entity( subTypeName );
+		ManagedDomainType<S> subManagedType = jpaMetamodel.resolveHqlEntityReference( subTypeName );
```

Comment:
usually subType is not specified as hibernate entity name, but as HQL entity name